x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.test.ts

```diff
@@ -0,0 +1,399 @@
+  let server = createMockServer();
+  let config = createMockConfig();
+  let getClients = clientsServiceMock.createGetScoped();
+  let clients = clientsServiceMock.createClients();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    server = createMockServer();
+    config = createMockConfig();
+    getClients = clientsServiceMock.createGetScoped();
+    clients = clientsServiceMock.createClients();
+
+    getClients.mockResolvedValue(clients);
+    clients.clusterClient.callAsCurrentUser.mockResolvedValue(getNonEmptyIndex());
+
+    importRulesRoute(server.route, config, getClients);
+  });
+
+  describe('status codes with actionsClient and alertClient', () => {
+    test('returns 200 when importing a single rule with a valid actionClient and alertClient', async () => {
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+      clients.alertsClient.get.mockResolvedValue(getResult());
+      clients.alertsClient.create.mockResolvedValue(getResult());
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1']);
+      const { statusCode } = await server.inject(getImportRulesRequest(requestPayload));
+      expect(statusCode).toEqual(200);
+    });
+
+    test('returns 404 if alertClient is not available on the route', async () => {
+      getClients.mockResolvedValue(omit('alertsClient', clients));
+      const { route, inject } = createMockServer();
+      importRulesRoute(route, config, getClients);
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1']);
+      const { statusCode } = await inject(getImportRulesRequest(requestPayload));
+      expect(statusCode).toEqual(404);
+    });
+
+    test('returns 404 if actionsClient is not available on the route', async () => {
+      getClients.mockResolvedValue(omit('actionsClient', clients));
+      const { route, inject } = createMockServer();
+      importRulesRoute(route, config, getClients);
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1']);
+      const { statusCode } = await inject(getImportRulesRequest(requestPayload));
+      expect(statusCode).toEqual(404);
+    });
+  });
+
+  describe('validation', () => {
+    test('returns reported error if index does not exist', async () => {
+      clients.clusterClient.callAsCurrentUser.mockResolvedValue(getEmptyIndex());
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+      clients.alertsClient.get.mockResolvedValue(getResult());
+      clients.alertsClient.create.mockResolvedValue(getResult());
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1']);
+      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
+      const parsed: ImportSuccessError = JSON.parse(payload);
+
+      expect(parsed).toEqual({
+        errors: [
+          {
+            error: {
+              message:
+                'To create a rule, the index must exist first. Index undefined-undefined does not exist',
```

Comment:
If you used the advice above for the configuration then this will become:

```ts
'To create a rule, the index must exist first. Index .siem-signals-default does not exist'
```

And look much more intentful.

Reply:
👍 updated!